### ISSUE TITLE
[Bug]: hide additional permission dialog on Android for file access when not needed

### ISSUE THREAD

**Issue by koenmuilwijk** [apivideo/api.video-flutter-uploader#24]
### Version

1.3.0

### Environment that reproduces the issue

* Android 14 device 

### Is it reproducible in the example application?

Not tested

### Reproduction steps

```
final uploadToken = '...';
final videoId = '...';
final video = await ImagePicker().pickVideo(
      source: ImageSource.gallery, 
      preferredCameraDevice: CameraDevice.front,
      maxDuration: Duration(minutes: 1),
    );
if (video != null) {
  await videouploader.ApiVideoUploader.uploadWithUploadToken(uploadToken, video.path, videoId: videoId);
}
```

### Expected result

When uploading a video an additional permission dialog is displayed on Android. However, in our use case we use [ImagePicker](https://pub.dev/packages/image_picker) for the user to select a file, so we don't need to ask for permission again.

Can we pass a direct link to the selected `XFile` or `File` instead of the path as string to prevent this?

Upload video without additional file permission request (already done by `ImagePicker().pickVideo(...)`).

### Actual result

An additional permission request is displayed upon the call to `uploadWithUploadToken` which doesn't serve any function.

![image](https://github.com/user-attachments/assets/98be2e89-7bb2-475f-8e70-2f82c76a7279)

### Additional context

Android 14 and higher.

**Comment by ThibaultBee** [comment]
Hi,

On Android 14 a new permission is required to read media.  See https://developer.android.com/about/versions/14/changes/partial-photo-video-access

It is just a wild guess but `ImagePicker` is requesting `Manifest.permission.READ_EXTERNAL_STORAGE` and the Flutter uploader is requesting `Manifest.permission.READ_MEDIA_VIDEO`. See https://github.com/apivideo/api.video-flutter-uploader/blob/main/android/src/main/kotlin/video/api/flutter/uploader/utils/Utils.kt#L10
You can check in your built `AndroidManifest.xml` that you got both permissions. Did you add a read permission (`Manifest.permission.READ_EXTERNAL_STORAGE` or `Manifest.permission.READ_MEDIA_VIDEO`) in your application `AndroidManifest.xml`?

`ImagePicker` is doing a [smarter move](https://github.com/flutter/packages/blob/main/packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerUtils.java). This could be a solution.

Also, we can't directly use `File` or `XFile` because under the hood we are using some native parts.

**Comment by koenmuilwijk** [comment]
Both permissions are included in the manifest. The weird thing is that it doesn't matter which option is selected in the permission dialog which is triggered by the call to `uploadWithUploadToken`, it will work regardless. Event when you select 'don't select more' it will just work.

Would it be an option to make the call to the permission dialog optional, so that we can pass in a bool indicating we already retrieved permission to access the file and there is no need to ask for it again?

The current flow is not very user friendly so it'd be great if this can be improved.

**Comment by ThibaultBee** [comment]
According to `Manifest.permission.READ_EXTERNAL_STORAGE` documentation:

> Note: Starting in API level 33, this permission has no effect. If your app accesses other apps' media files, request one or more of these permissions instead: [READ_MEDIA_IMAGES](https://developer.android.com/reference/android/Manifest.permission#READ_MEDIA_IMAGES), [READ_MEDIA_VIDEO](https://developer.android.com/reference/android/Manifest.permission#READ_MEDIA_VIDEO), [READ_MEDIA_AUDIO](https://developer.android.com/reference/android/Manifest.permission#READ_MEDIA_AUDIO). Learn more about the [storage permissions](https://developer.android.com/training/data-storage/shared/media#storage-permission) that are associated with media files.

I don't understand why `ImagePicker` requests `Manifest.permission.READ_EXTERNAL_STORAGE` and not `Manifest.permission.READ_MEDIA_VIDEO`.

I don't remember having the same issue in the example app and we are using `ImagePicker` as well.
Do you manually add permissions in your `AndroidManifest.xml`?

**Comment by koenmuilwijk** [comment]
I just tested with the example app and it basically has the same issue.

1. Start example Flutter app
2. Select a video ('video-a.mp4') with the picker (no permission dialog is shown for me)
3. Now it will start upload and call to `uploadWithUploadToken` will trigger permission dialog

Few things can happen:
1. Select Allow all: it will work
2. Select Allow limited: it will still work, EVEN when you select a different video file ('video-b.mp4')!
3. Select Don't select more and it will give an error toast, but my gut feeling because `uploadWithUploadToken` will not continue, but my gut feeling is that it would have worked if the check could be bypassed

Can we make the permission check optional? That'd prevent this issue and make user flow much better.

**Comment by ThibaultBee** [comment]
Oh sorry, I understood you got the permission request twice but that's not that.
You just don't want any permission dialog. Right?

**Comment by koenmuilwijk** [comment]
Correct, I want a way to prevent it popping up when it's not needed.

**Comment by ThibaultBee** [comment]
Oh, the file picker grants a temporary access to the file. I did not know. Indeed there should be a way to NOT request the permission (starting from Android 13). Maybe there is some way to check that we have access to the file, that. would make a simpler API.
